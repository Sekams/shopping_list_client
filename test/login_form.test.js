import React from 'react';
import LoginForm from '../src/components/login_form'
import Spinner from '../src/components/spinner'

test("renders the login_form component", () => {
    const wrapper = shallow(
        <LoginForm />
    );
    const spinner = shallow(
        <Spinner />
    );
    wrapper.instance()._spinner = spinner.instance();
    wrapper.instance()._spinner.showSpinner = jest.fn();
    global.showSpinner(wrapper.instance());
    global.dismissSpinner(wrapper.instance());
    wrapper.props().children[1].ref();
    expect(wrapper).toMatchSnapshot();
});

test("handles inputs", () => {
    const term = 'food';
    const wrapper = shallow(
        <LoginForm />
    );
    wrapper.find('input').map((input) => {
        input.simulate(('change'), {
            target: { term },
        });
    });
    wrapper.instance().handleLogin({ preventDefault() {} })
    expect(wrapper).toMatchSnapshot();
});

test("handles login", () => {
    const term = 'food';
    fetch.mockResponseOnce(JSON.stringify(
        {
            "access_token": "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJleHAiOjE1MTM3MDc0OTEsImlhdCI6MTUxMzYyMTA5MSwic3ViIjoxfQ.9jyOGbBDTYmcvWELtmZetXyn3euOBQGVLnnKog9lbUs",
            "message": "You logged in successfully.",
            "status": "success",
        }),
        {
            status: 200,
            ok: true,
        }
    );
    const wrapper = shallow(
        <LoginForm />
    );
    wrapper.instance().setState({
        username: 'user',
        password: '123456'
    });
    wrapper.update();
    wrapper.instance().handleLogin({ preventDefault() {} })
    expect(wrapper).toMatchSnapshot();
});

test("handles unsuccessful login", () => {
    const term = 'food';
    fetch.mockResponseOnce(JSON.stringify(
        {
            "message": "Login not successful.",
            "status": "fail",
        }),
        {
            status: 200,
            ok: true,
        }
    );
    const wrapper = shallow(
        <LoginForm />
    );
    wrapper.instance().setState({
        username: 'user',
        password: '123456'
    });
    wrapper.update();
    wrapper.instance().handleLogin({ preventDefault() {} })
    expect(wrapper).toMatchSnapshot();
});