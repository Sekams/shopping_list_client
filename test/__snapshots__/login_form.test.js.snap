// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handles inputs 1`] = `
<div>
  <NavBar
    page="login"
  />
  <div
    className="login-wrapper"
  >
    <div
      className="card login-card drop-shadow"
    >
      <h3
        className="title"
      >
        Login
      </h3>
      <form
        onSubmit={[Function]}
      >
        <input
          className="form-input form-control"
          onChange={[Function]}
          placeholder="Username"
          type="text"
        />
        <input
          className="form-input form-control"
          onChange={[Function]}
          placeholder="Password"
          type="password"
        />
        <button
          className="btn btn-form-submit btn-login-submit"
          type="submit"
        >
          Login
        </button>
      </form>
    </div>
  </div>
</div>
`;

exports[`handles login 1`] = `
<div>
  <NavBar
    page="login"
  />
  <div
    className="login-wrapper"
  >
    <div
      className="card login-card drop-shadow"
    >
      <h3
        className="title"
      >
        Login
      </h3>
      <form
        onSubmit={[Function]}
      >
        <input
          className="form-input form-control"
          onChange={[Function]}
          placeholder="Username"
          type="text"
          value="user"
        />
        <input
          className="form-input form-control"
          onChange={[Function]}
          placeholder="Password"
          type="password"
          value="123456"
        />
        <button
          className="btn btn-form-submit btn-login-submit"
          type="submit"
        >
          Login
        </button>
      </form>
    </div>
  </div>
</div>
`;

exports[`handles unsuccessful login 1`] = `
<div>
  <NavBar
    page="login"
  />
  <div
    className="login-wrapper"
  >
    <div
      className="card login-card drop-shadow"
    >
      <h3
        className="title"
      >
        Login
      </h3>
      <form
        onSubmit={[Function]}
      >
        <input
          className="form-input form-control"
          onChange={[Function]}
          placeholder="Username"
          type="text"
          value="user"
        />
        <input
          className="form-input form-control"
          onChange={[Function]}
          placeholder="Password"
          type="password"
          value="123456"
        />
        <button
          className="btn btn-form-submit btn-login-submit"
          type="submit"
        >
          Login
        </button>
      </form>
    </div>
  </div>
  <SnackBar
    class="success-snackbar"
    message="You logged in successfully."
  />
</div>
`;

exports[`renders the login_form component 1`] = `
<div>
  <NavBar
    page="login"
  />
  <div
    className="login-wrapper"
  >
    <div
      className="card login-card drop-shadow"
    >
      <h3
        className="title"
      >
        Login
      </h3>
      <form
        onSubmit={[Function]}
      >
        <input
          className="form-input form-control"
          onChange={[Function]}
          placeholder="Username"
          type="text"
          value=""
        />
        <input
          className="form-input form-control"
          onChange={[Function]}
          placeholder="Password"
          type="password"
          value=""
        />
        <button
          className="btn btn-form-submit btn-login-submit"
          type="submit"
        >
          Login
        </button>
      </form>
    </div>
  </div>
</div>
`;
